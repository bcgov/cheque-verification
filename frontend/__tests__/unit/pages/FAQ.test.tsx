import { describe, it, expect } from "vitest";
import { screen } from "@testing-library/react";
import { renderWithProviders } from "../../helpers/testHelpers";
import Faq from "../../../src/pages/FAQ";

describe("FAQ Page", () => {
  it("renders the main FAQ heading", () => {
    renderWithProviders(<Faq />);

    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveTextContent("Frequently Asked Questions");
  });

  it("applies correct BC Government styling to the header", () => {
    renderWithProviders(<Faq />);

    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveStyle("font-size: 28px");
    expect(heading).toHaveStyle("font-weight: 600");
  });

  it("renders all FAQ sections with proper headings", () => {
    renderWithProviders(<Faq />);

    // Check for all main FAQ sections using actual content
    expect(screen.getByText("Question 1?")).toBeInTheDocument();

    expect(
      screen.getByText("How often is the data updated?")
    ).toBeInTheDocument();

    expect(
      screen.getByText("What information do I need to verify a cheque?")
    ).toBeInTheDocument();
  });

  it("contains correct information about data updates", () => {
    renderWithProviders(<Faq />);

    expect(
      screen.getByText(
        /Cheque verification results are updated nightly at 3:00 AM PT/
      )
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        /may not appear in the system until the next business day/
      )
    ).toBeInTheDocument();
  });

  it("lists all required information for cheque verification", () => {
    renderWithProviders(<Faq />);

    expect(
      screen.getByText(/Cheque Number \(found on your cheque\)/)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Payment Issue Date \(the date the cheque was issued\)/)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Cheque Amount \(the dollar amount on the cheque\)/)
    ).toBeInTheDocument();
  });

  it("explains different cheque statuses", () => {
    renderWithProviders(<Faq />);

    expect(screen.getByText("Valid Cheque Statuses:")).toBeInTheDocument();

    // Check for status explanations (these would be in the updated content)
    expect(
      screen.getByText(/The cheque has been created and is ready to be cashed/)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/The cheque has been successfully processed and cashed/)
    ).toBeInTheDocument();
  });

  it("provides troubleshooting information", () => {
    renderWithProviders(<Faq />);

    expect(
      screen.getByText(
        /your cheque was issued within the last 24 hours, it may not appear in the system/
      )
    ).toBeInTheDocument();
  });

  it("includes contact information for help", () => {
    renderWithProviders(<Faq />);

    expect(
      screen.getByText(
        /If you have additional questions or need assistance, please contact Example contact/
      )
    ).toBeInTheDocument();
  });

  it("has proper semantic HTML structure", () => {
    renderWithProviders(<Faq />);

    // Should have main landmark
    const main = screen.getByRole("main");
    expect(main).toBeInTheDocument();

    // Should have proper heading hierarchy
    const h1 = screen.getByRole("heading", { level: 1 });
    expect(h1).toBeInTheDocument();

    const h2Elements = screen.getAllByRole("heading", { level: 2 });
    expect(h2Elements.length).toBeGreaterThan(0);

    const h3Elements = screen.getAllByRole("heading", { level: 3 });
    expect(h3Elements.length).toBeGreaterThan(0);
  });

  it("uses proper list structures for information", () => {
    renderWithProviders(<Faq />);

    // Should contain unordered lists
    const lists = screen.getAllByRole("list");
    expect(lists.length).toBeGreaterThan(0);

    // Should contain list items
    const listItems = screen.getAllByRole("listitem");
    expect(listItems.length).toBeGreaterThan(0);
  });

  it("applies correct container styling", () => {
    renderWithProviders(<Faq />);

    const main = screen.getByRole("main");
    // Check for key styling properties that matter for the layout
    expect(main).toHaveStyle("width: 100%");
    expect(main).toHaveStyle("max-width: 800px");
  });

  it("has proper card-like styling for the content container", () => {
    renderWithProviders(<Faq />);

    // Check that elements have appropriate styling attributes
    expect(screen.getByRole("main")).toBeInTheDocument();

    // Since inline styles are complex objects, just verify the container structure exists
    const container = screen
      .getByText("Frequently Asked Questions")
      .closest("div");
    expect(container).toBeInTheDocument();
  });

  it("has BC Government header styling with blue background and gold border", () => {
    renderWithProviders(<Faq />);

    // Find the header element by its content and verify it has the right styling attributes
    const heading = screen.getByText("Frequently Asked Questions");
    expect(heading).toBeInTheDocument();

    // Get the parent div that should have the blue background and gold border
    const headerDiv = heading.closest(
      'div[style*="background-color: var(--bcgov-blue)"]'
    );
    expect(headerDiv).toBeInTheDocument();

    // Verify the element has the expected style attributes
    // React expands the borderBottom shorthand into individual properties
    expect(headerDiv).toBeDefined();
    if (headerDiv) {
      const styleAttr = headerDiv.getAttribute("style");
      expect(styleAttr).toContain("background-color: var(--bcgov-blue)");
      // Check for the border properties that React actually renders
      expect(styleAttr).toContain("border-bottom-width: var(--bcgov-gold)");
      expect(styleAttr).toContain("border-bottom-color: var(--bcgov-gold)");
    }
  });

  it("uses consistent text styling throughout", () => {
    renderWithProviders(<Faq />);

    // Check that the main structure has proper headings and text
    const headings = screen.getAllByRole("heading", { level: 2 });
    expect(headings.length).toBeGreaterThan(0);
  });

  it("displays main heading correctly", () => {
    renderWithProviders(<Faq />);

    // Check for heading structure and text
    const heading = screen.getByText("Frequently Asked Questions");
    expect(heading).toBeInTheDocument();
  });

  it('has highlighted "Need More Help" section', () => {
    renderWithProviders(<Faq />);

    expect(screen.getByText(/Need More Help\?/)).toBeInTheDocument();

    expect(
      screen.getByText(
        /If you have additional questions or need assistance, please contact Example contact/
      )
    ).toBeInTheDocument();
  });
  it("matches snapshot for visual consistency", () => {
    const { container } = renderWithProviders(<Faq />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
