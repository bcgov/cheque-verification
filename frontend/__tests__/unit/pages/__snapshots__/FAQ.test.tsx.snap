// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FAQ Page > matches snapshot for visual consistency 1`] = `
<main
  style="width: 100%; max-width: 800px; margin: 0px auto; padding: 24px 16px 200px;"
>
  <div
    style="background-color: var(--bcgov-background-white); border-radius: 6px; box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1); overflow: hidden; border: 1px solid; border-width: var(--bcgov-border-light); border-style: var(--bcgov-border-light); border-color: var(--bcgov-border-light);"
  >
    <div
      style="background-color: var(--bcgov-blue); padding: 24px; display: flex; align-items: center; border-bottom-width: var(--bcgov-gold); border-bottom-style: var(--bcgov-gold); border-bottom-color: var(--bcgov-gold);"
    >
      <h1
        style="color: var(--bcgov-text-white); font-size: 28px; font-family: BCSans, sans-serif; font-weight: 600; margin: 0px;"
      >
        Frequently Asked Questions (FAQ)
      </h1>
    </div>
    <div
      style="padding: 32px; font-family: BCSans, sans-serif; line-height: 1.6;"
    >
      <div
        style="margin-bottom: 32px;"
      >
        <h2
          style="color: var(--bcgov-text-primary); font-size: 20px; font-weight: 600; margin-bottom: 12px; padding-bottom: 8px; border-bottom-width: var(--bcgov-gold); border-bottom-style: var(--bcgov-gold); border-bottom-color: var(--bcgov-gold);"
        >
          How often is the information Refreshed?
        </h2>
        <ul
          style="color: var(--bcgov-text-primary); font-size: 16px; margin-left: 24px; margin-bottom: 16px;"
        >
          <li
            style="margin-bottom: 8px;"
          >
            Cheque information is refreshed daily at 6AM.
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            This information captures data from previous day, information cut off at 5pm.
          </li>
        </ul>
      </div>
      <div
        style="margin-bottom: 32px;"
      >
        <h2
          style="color: var(--bcgov-text-primary); font-size: 20px; font-weight: 600; margin-bottom: 12px; padding-bottom: 8px; border-bottom-width: var(--bcgov-gold); border-bottom-style: var(--bcgov-gold); border-bottom-color: var(--bcgov-gold);"
        >
          Cheque #: Enter in top right Cheque # Information.
        </h2>
        <div
          style="display: flex; justify-content: center; margin-bottom: 16px;"
        >
          <img
            alt="Sample cheque showing where to find the cheque number in the top right corner"
            src="/chequenumber.png"
            style="width: 100%; max-width: 100px; height: auto; border: 1px solid; border-radius: 4px; border-width: var(--bcgov-border-light); border-style: var(--bcgov-border-light); border-color: var(--bcgov-border-light);"
          />
        </div>
      </div>
      <div
        style="margin-bottom: 32px;"
      >
        <h2
          style="color: var(--bcgov-text-primary); font-size: 20px; font-weight: 600; margin-bottom: 12px; padding-bottom: 8px; border-bottom-width: var(--bcgov-gold); border-bottom-style: var(--bcgov-gold); border-bottom-color: var(--bcgov-gold);"
        >
          Can you tell me why a cheque was issued or details about a client's payment history?
        </h2>
        <p
          style="color: var(--bcgov-text-primary); font-size: 16px; margin-bottom: 8px;"
        >
          No. Due to privacy regulations, we cannot provide details such as:
        </p>
        <ul
          style="color: var(--bcgov-text-primary); font-size: 16px; margin-left: 24px; margin-bottom: 16px;"
        >
          <li
            style="margin-bottom: 8px;"
          >
            Why a cheque was issued
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            How many cheques a client is supposed to receive
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            Why a cheque was stopped
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            Whether a client has a history of stopped cheques
          </li>
        </ul>
      </div>
      <div
        style="margin-bottom: 32px;"
      >
        <h2
          style="color: var(--bcgov-text-primary); font-size: 20px; font-weight: 600; margin-bottom: 12px; padding-bottom: 8px; border-bottom-width: var(--bcgov-gold); border-bottom-style: var(--bcgov-gold); border-bottom-color: var(--bcgov-gold);"
        >
          What are the common verification phone numbers for agencies?
        </h2>
        <p
          style="color: var(--bcgov-text-primary); font-size: 16px; margin-bottom: 8px;"
        >
          Here is a quick reference list:
        </p>
        <ul
          style="color: var(--bcgov-text-primary); font-size: 16px; margin-left: 24px; margin-bottom: 16px;"
        >
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              MCFD Cheques:
            </strong>
             250-356-8139
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Day Care Subsidy Cheques:
            </strong>
             1-888-338-6622
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Family Maintenance Cheques:
            </strong>
             1-800-663-9666 or 604-678-5670
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Community Living Cheques:
            </strong>
             604-664-0784
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Province of BC General / MSP / Payroll:
            </strong>
             
            1-888-587-7114
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Enquiry BC:
            </strong>
             1-800-663-7867
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Province of BC Payment Enquiry (SSBC CAS Business Services):
            </strong>
             
            1-800-663-7867
          </li>
          <li
            style="margin-bottom: 8px;"
          >
            <strong>
              Federal Cheques:
            </strong>
             1-866-552-8034
          </li>
        </ul>
      </div>
    </div>
  </div>
</main>
`;
